lib/cell-manager.coffee

```diff
@@ -1,57 +1,26 @@
-_ = require 'lodash'
-
 module.exports = CellManager =
-    _setBreakpointDecorationAt: (editor, row) ->
-        marker = editor.markBufferPosition
-            row: row
-            column: 0
-        ,
-            invalidate: 'never'
-
-        editor.decorateMarker marker,
-            type: 'line-number'
-            class: 'breakpoint'
-
-    _getBreakprointDecorationAt: (editor, row) ->
-        decorations = editor
-            .getLineNumberDecorations({class: 'breakpoint'})
-            .filter (decoration) ->
-                return decoration.marker.getStartBufferPosition().row is row
-        return decorations[0]
-
-    removeAllBreakpoints: ->
-        editor = atom.workspace.getActiveTextEditor()
-        decorations = editor.getLineNumberDecorations({class: 'breakpoint'})
-        for decoration in decorations
-            decoration.marker.destroy()
-
-    toggleBreakpoint: ->
-        editor = atom.workspace.getActiveTextEditor()
-        row = editor.getLastCursor().getBufferRow()
-        decoration = CellManager._getBreakprointDecorationAt editor, row
-
-        if decoration
-            decoration.marker.destroy()
-            return
-
-        CellManager._setBreakpointDecorationAt editor, row
-
     getCurrentCell: ->
         editor = atom.workspace.getActiveTextEditor()
+        buffer = editor.getBuffer()
         row = editor.getLastCursor().getBufferRow()
-        decorations = editor.getLineNumberDecorations({class: 'breakpoint'})
+        scope = editor.scopeDescriptorForBufferPosition([row, 0])
 
         start = 0
         end = editor.getLastBufferRow()
-        for decoration in decorations
-            decorationRow = decoration.marker.getStartBufferPosition().row
 
-            if decorationRow >= row
-                if (decorationRow < end)
-                    end = decorationRow
-            else
-                if (decorationRow >= start)
-                    start = decorationRow + 1
+        {commentStartString, commentEndString} =
+            editor.languageMode.commentStartAndEndStringsForScope(scope)
+
+        if commentStartString
+            regex = new RegExp(commentStartString + '%%|<codecell>|In\[[0-9 ]+\]:')
```

Comment:
Should we enforce that the comment starts a line?

I would find the below example confusing:

```js
a = 1; # %% In[1]:
console.log(a);

a += 1; # %% In[2]:
console.log(a);
```